src/app/public/modules/definition-list/definition-list.component.ts

```diff
@@ -34,9 +72,65 @@ export class SkyDefinitionListComponent {
  */
   @Input()
   public set defaultValue(value: string) {
-    this.service.defaultValue.next(value);
+    this.definitionListService.defaultValue.next(value);
   }
 
+  @Input()
+  public mode: SkyDefinitionListMode = 'default';
+
+  @Input()
+  public orientation: SkyDefinitionListOrientation = 'vertical';
+
+  public isMobile: boolean = false;
+
+  // Use Observable to allow template to get at content of its grandchildren.
+  public templateStream: Subject<QueryList<SkyDefinitionListContentComponent>> =
+    new Subject<QueryList<SkyDefinitionListContentComponent>>();
+
+  @ContentChildren(SkyDefinitionListContentComponent)
+  private contentComponents: QueryList<SkyDefinitionListContentComponent>;
+
+  @ContentChild(SkyDefinitionListHeadingComponent)
+  private headerComponent: SkyDefinitionListHeadingComponent;
+
+  @ViewChild('definitionListElement', {
+    read: ElementRef,
+    static: true
+  })
+  private elementRef: ElementRef;
+
+  private _labelWidth: string;
+
   constructor(
-    public service: SkyDefinitionListService) { }
+    private adapterSerivce: SkyDefinitionListAdapterService,
+    private changeDetector: ChangeDetectorRef,
+    private definitionListService: SkyDefinitionListService
+  ) { }
+
+  public ngAfterContentInit(): void {
+    if (this.headerComponent) {
+      console.warn(
+        '[Deprecation warning] Do not use the `<sky-definition-list-heading>` component in your template (this will be a breaking change in the next major version release).\n' +
+        'Instead, include a header above the defintion list component that uses a SKY UX' +
+        ' supported class:\n' +
+        '<h3 class="sky-subsection-heading sky-margin-stacked-compact">\n  My list' +
+        '\n</h3>\n<sky-definition-list>\n  ...\n</sky-definition-list>'
+      );
+    }
+
+    setTimeout(() => {
+      this.templateStream.next(this.contentComponents);
```

Comment:
Do you mind providing a reason around using a stream instead of using a `set` method on the `contentComponents` property?

Reply:
Originally, this was to capture changes from the grandchildren components. Now that things have shifted with how I'm accessing that content, I can abandon this stream and just use `contentComponents`. I've made the change and also restructured a test to cover changes pushed to that `QueryList`.